plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,59 @@
+package io.jenkins.plugins.casc;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.misc.Util;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static io.jenkins.plugins.casc.SchemaGeneration.generateSchema;
+import static org.junit.Assert.fail;
+
+public class SchemaGenerationTest {
+
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    public void validSchemaShouldSucceed() throws Exception {
+        JSONObject schemaObject = generateSchema();
+        JSONObject jsonSchema = new JSONObject(
+            new JSONTokener(schemaObject.toString()));
+        String yamlStringContents = Util.toStringFromYamlFile(this, "validSchemaConfig.yml");
+        JSONObject jsonSubject = new JSONObject(
+            new JSONTokener(Util.convertToJson(yamlStringContents)));
+        Schema schema = SchemaLoader.load(jsonSchema);
+        schema.validate(jsonSubject);
+    }
+
+    @Test
+    public void invalidSchemaShouldNotSucceed() throws Exception {
+
+        JSONObject schemaObject = generateSchema();
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        JsonParser jsonParser = new JsonParser();
+        JsonElement jsonElement = jsonParser.parse(schemaObject.toString());
+        String prettyJsonString = gson.toJson(jsonElement);
+        JSONObject jsonSchema = new JSONObject(
+            new JSONTokener(prettyJsonString));
+        String yamlStringContents = Util.toStringFromYamlFile(this, "invalidSchemaConfig.yml");
+        JSONObject jsonSubject = new JSONObject(
+            new JSONTokener(Util.convertToJson(yamlStringContents)));
+        Schema schema = SchemaLoader.load(jsonSchema);
+        try {
+            schema.validate(jsonSubject);
```

Comment:
Please make an assertion at least

Reply:
Its a void so probably put an `Assert.fail(e.getMessage());` in the try catch :)